host-bin/edit-chroot

```diff
@@ -385,8 +434,33 @@ for NAME in "$@"; do
         fi
         echo -n "  Restoring $src to $CHROOT..." 1>&2
         mkdir -p "$CHROOT"
-        spinner 1 tar --checkpoint=200 --checkpoint-action=exec=echo \
-            --one-file-system -xaf "$src" -C "$CHROOT" --strip-components=1
+        if [ -f "$src.part01" ]; then
+            # Detect the type of compression before sending it through a fifo
+            for tarparam in -z -j -J -Z --no-auto-compress fail; do
+                if [ "$tarparam" = 'fail' ]; then
+                    error 2 "Unable to detect compression method of $src"
+                elif tar $tarparam --test-label -f "$src" >/dev/null 2>&1; then
```

Comment:
Tried to estimate minimum block size:

```
cat big.tar | dd count=1 bs=1K iflag=fullblock | tar --test-label -v; echo $?
gzip -c big.tar | dd count=1 bs=50K iflag=fullblock | tar --test-label -vz; echo $?
bzip2 -c big.tar | dd count=1 bs=900K iflag=fullblock | tar --test-label -vj; echo $?
xz -c big.tar | dd count=1 bs=20K iflag=fullblock | tar --test-label -vJ; echo $?
```

It depends on the compression ratio of the archive, but bzip2 has a clear minimum at 900K (it compresses blocks of 900K of input).

Maybe 10M to be safe? (lzma -6 in memory dictionary is 8MiB) I mean, it's not like somebody is going to backup to floppy disks (right???!?)